portfolio/src/main/webapp/script.js

```diff
@@ -115,28 +116,90 @@ function createCommentElement(comment) {
+  window.location.replace("index.html?maxComments="+ document.getElementById('quantity').value );
+}
+
+/**Set the default value to maxComments when user doesn't give any input or when the page first loads. */
+function getMaxComments() {
+    const defaultValue = 15;
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    if(urlParams.get('maxComments') == null)
+        return defaultValue;
+    else
+        return(urlParams.get('maxComments'));
 }
 
-// Close the dropdown if the user clicks outside of it
-window.onclick = function(event) {
-  if (!event.target.matches('.commentSettings')) {
-    var dropdowns = document.getElementsByClassName("dropdown-content");
-    var i;
-    for (i = 0; i < dropdowns.length; i++) {
-      var openDropdown = dropdowns[i];
-      if (openDropdown.classList.contains('show')) {
-        openDropdown.classList.remove('show');
-      }
+/**Makes the form visible and adds the 'action' to it by fetching 
+the url(that the form needs to post to) from the servlet */
+function fetchBlobstoreUrlAndShowForm() {
+  fetch('/blobstore-upload-url')
+      .then((response) => response.text())
+      .then((imageUploadUrl) => {
+        const messageForm = document.getElementById('upload-images');
+        messageForm.action = imageUploadUrl;
+        messageForm.classList.remove('hidden');
+      });
+}
+
+/**Fetching all image urls stored in datastore and presenting them on screen along with date/time and its message */
+function getAndShowImages(){
+  const imageListElement = document.getElementById('images-container');
+
+  fetch("/my-image-servlet").then((response) => response.json()).then((imageDetails) => {
+    if(imageDetails.length==0) {
+        imageListElement.innerText = "No images";
+    }
+    else {
+        imageDetails.forEach((imageDetail) => {
+            imageListElement.appendChild(createImageElement(imageDetail));
+        });
     }
-  }
+  })
+}
+
+/**Creating an image element */
+function createImageElement(imageDetail) {
+  const imageElement = document.createElement('li');
+  imageElement.className = 'image';
+
+  const myImg = document.createElement("IMG");
+  myImg.setAttribute("src", imageDetail.imageUrl);
+  myImg.setAttribute("width", "250");
+  myImg.setAttribute("height", "300");
+  myImg.setAttribute("alt", imageDetail.id);
+
+  const myExpandedImg = document.createElement("A");
+  myExpandedImg.setAttribute("href", imageDetail.imageUrl);
+  myExpandedImg.appendChild(myImg);
+        
+  const messageElement = document.createElement('p');
+  messageElement.innerText = imageDetail.message;
+  messageElement.className = 'caption';
+
+  const timeElement = document.createElement('p');
+  timeElement.className = 'date';
+  const date = new Date(imageDetail.timestamp).toLocaleDateString("en-US");
+  const time = new Date(imageDetail.timestamp).toLocaleTimeString("en-US"); 
+  timeElement.innerHTML = "&#128344;"+date+"&nbsp;"+time;
```

Comment:
Consider writing `` `🕘${ date }&nbsp;${ time }` ``
It's fine to use unicode characters in code because this way we see what will be displayed on the page.

Reply:
Okay will make sure to do that from now on. Thanks for letting me know :)